perch/model.py

```diff
@@ -111,16 +115,25 @@ def id(self, value):
 
     @coroutine
     def validate(self):
-        """Validate the resource using it's voluptuous schema"""
+        """Validate the resource using its voluptuous schema"""
         try:
-            # update _resource so have default values from the schema
+            # update _resource to have default values from the schema
             self._resource = self.schema(self._resource)
         except MultipleInvalid as e:
             errors = [format_error(err, self.resource_type) for err in e.errors]
             raise exceptions.ValidationError({'errors': errors})
 
         yield self.check_unique()
 
+    @classmethod
```

Comment:
Why does this need to be a `classmethod`? If it was an instance method you could just do `self.schema` instead of requiring the `schema` parameter